### ISSUE TITLE
Write Webapp E2E with Cypress: "MM-T1805 No infinite loading spinner on Select Team page"

### ISSUE THREAD

**Issue by saturninoabril** [mattermost/mattermost-server#18178]
See our [documentation for Webapp end-to-end testing with Cypress](https://developers.mattermost.com/contribute/webapp/end-to-end-tests/) for reference.

<article>
<h1>MM-T1805 No infinite loading spinner on Select Team page</h1>
<div>
<div>
<h3>Steps </h3>1. Create an account on a server with very few teams, like https://rc.test.mattermost.com<br>2. Join all available teams teams<br>3. Go to {servername}/select_team (e.g. <a href="https://rc.test.mattermost.com/select_team">https://rc.test.mattermost.com/select_team</a>)<br>4. Observe the Select Team page<h3>Expected</h3><span data-sheets-userformat='{"2":769,"3":{"1":0},"11":4,"12":0}' data-sheets-value='{"1":2,"2":"No infinite loading spinner is shown below the \"No teams available to join\" message"}'>No infinite loading spinner is shown below the "No teams available to join" message</span><hr>
</div>
</div>
</article>

**Test Folder:** ``/cypress/integration/multi-team_+_dms``
**Test code arrangement:**
```
describe('Multi-Team + DMs', () => {
    it('MM-T1805 No infinite loading spinner on Select Team page', () => {
        // code
    });
});
```

Notes:
1. Do not add ``@prod`` label in a spec file
  - If you're writing script into a newly created test file, ``@prod`` label should not be included.
  - If you're adding script into an existing test file, ``@prod`` label should removed.
2. Use [queries from testing-library](https://testing-library.com/docs/dom-testing-library/api-queries) whenever possible. We share the same philosophy as the [testing-library](https://testing-library.com/) when doing UI automation like "Interact with your app the same way as your users" and so, please follow their guidelines especially when querying an element. 

If you're interested, please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

**Comment by poflankov** [comment]
Can I work it?

**Comment by saturninoabril** [comment]
@poflankov Sure and thanks! I've updated the test not to mention other server name. Test could be made against the local server.

**Comment by poflankov** [comment]
How can I simulate the situation of appearing the spinner here?

**Comment by saturninoabril** [comment]
The spinner will appear during fetching of the next page of teams available for joining. To simulate it manually, create more than 30 teams (that allows open invite), visit `/select_team` page, then scroll down the list of teams. Observe that the spinner shows up until the newly fetched data is rendered in the list. So, it's a combination of UI action (scrolling), scroll position (near the bottom of the list) and if there's more team to fetch. See code at https://github.com/mattermost/mattermost-webapp/blob/master/components/common/infinite_scroll.tsx#L145

**Comment by poflankov** [comment]
Thanks for clarifying. But I've got a question: step 1 says "very few teams". Are you sure it's a case for long list of teams? Maybe it's different test case?

**Comment by saturninoabril** [comment]
I only responded to your question and I thought you'd like to see what kind of spinner is to be observed there.
> How can I simulate the situation of appearing the spinner here?

The test is not intended for a long list of teams but exactly on the stated steps and expectation above. When no more team to join, no spinner should be visible on `/select_team` page. See screenshot of the past bug and [the ticket](https://mattermost.atlassian.net/browse/MM-23042) where this test case is based upon.
![image-2020-03-05-09-27-04-575](https://user-images.githubusercontent.com/5334504/135243884-1d382455-4eaa-45e2-a667-e6798bf58918.png)
e

**Comment by poflankov** [comment]
Ah, thanks, got it now.
I've already made the test in fact, which exactly checks the absence of the spinner.
And I wanted to test my test so it breaks when something makes the spinner appear.
```js
    it.only('MM-T1805 No infinite loading spinner on Select Team page', () => {
        cy.visit('/select_team');
        cy.get('#loadingSpinner').should('not.exist');
    });
```
I mean if the spinner should appear - will this test do the task?
E.g. if the spinner somehow appears after the `cy.get('#loadingSpinner').should('not.exist')` assertion.

**Comment by saturninoabril** [comment]
Yeah, something like that, except it's using other loading indicator ([LoadingScreen](https://github.com/mattermost/mattermost-webapp/blob/master/components/loading_screen.tsx)). So, it could be:
```javascript
cy.get('.loading-screen').should('not.exist');
```
